File: src/main/java/com/polaris/model/Operario.java
package com.polaris.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.fasterxml.jackson.annotation.JsonIgnore;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "operario")
public class Operario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(length = 100, nullable = false, unique = true)
    private String correo;

    @Column(length = 100, nullable = false)
    private String contrasena;

    @Column(length = 80, nullable = false)
    private String nombre;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_admin", nullable = false)
    private Administrador administrador;
}

File: frontend/src/app/models/operario.ts
export interface Operario {
  id: number;
  correo: string;
  contrasena: string;
  nombre: string;
  administradorId: number;
}

export class OperarioModel implements Operario {
  id: number = 0;
  correo: string = '';
  contrasena: string = '';
  nombre: string = '';
  administradorId: number = 0;

  constructor(data?: Partial<Operario>) {
    Object.assign(this, data);
  }
}